src/components/Table/utils/TableState.ts

```diff
@@ -105,7 +105,8 @@ export class TableState {
   }
 
   loadSelected(rows: GridDataRow<any>[]): void {
-    const selectedRows = rows.filter((row) => row.initSelected);
+    const childRows = rows.flatMap((r) => (r.children ? flattenRows(r.children) : []));
+    const selectedRows = [...rows, ...childRows].filter((row) => row.initSelected);
```

Comment:
Can you add a unit test for this behavior as well (or instead)? The story is fine, but this sounds like a behavior change more than a visual change, and so a unit test is better for regression coverage.

Reply:
Hi @stephenh 

I included a unit test with this scenario with the commit

Reply:
Hi @stephenh 

Unit test added :), thank you!